ods_ci/utils/scripts/fetch_tests.py

```diff
@@ -20,39 +21,58 @@ def get_repository(test_repo):
     If $test_repo is a local path, the function checks the path exists.
     """
     repo_local_path = "./ods_ci/ods-ci-temp"
+    cloned = False
     if "http" in test_repo or "git@" in test_repo:
         print("Cloning repo ", test_repo)
-        ret = execute_command(f"git clone {test_repo} {repo_local_path}")
-        if "error" in ret.lower():
-            # actual error gets printed during "execute_command"
-            raise Exception("Failed to clone the given repository")
+        cloned = True
+        execute_command(f"git clone {test_repo} {repo_local_path}")
     elif not os.path.exists(test_repo):
         raise FileNotFoundError(f"local path {test_repo} was not found")
     else:
         print("Using local repo ", test_repo)
         repo_local_path = test_repo
-    return repo_local_path
+    return repo_local_path, cloned
 
 
 def checkout_repository(ref):
     """
     Checkouts the repository at current directory to the given branch/commit ($ref)
     """
-    ret = execute_command(f"git checkout {ref}")
-    if "error" in ret.lower():
-        # actual error gets printed during "execute_command"
-        raise Exception(f"Failed to checkout to the given branch/commit {ref}")
-    ret = execute_command("git checkout")
+    execute_command(f"git checkout {ref}")
+    execute_command("git checkout")
 
 
-def extract_test_cases_from_ref(repo_local_path, ref):
+def get_branch(ref_to_exclude, selector_attribute):
+    """
+    List the remote branches and sort by last commit date (ASC order), exclude $ref_to_exclude and get latest
+    """
+    ref_to_exclude_esc = ref_to_exclude.replace("/", r"\/")
+    if "master" in ref_to_exclude or "main" in ref_to_exclude:
+        cmd = f"git branch -r --sort={selector_attribute} | grep releases/"
+    else:
+        cmd = rf"git branch -r --sort={selector_attribute} | grep releases/  | sed  's/.*{ref_to_exclude_esc}$/current/g' |  grep -zPo '[\S\s]+(?=current)'"
+    ret = execute_command(cmd)
+    branches = ret.split(" ")
+    ret = branches[-1].split("\x00")[0].strip().replace("\n", "")
```

Comment:
maybe "branch" instead of generic "ret"
could you please give an example where we need the strip and replace?  
```suggestion
    branch = branches[-1].split("\x00")[0].strip().replace("\n", "")
```

Reply:
without replace: 
![image](https://github.com/red-hat-data-services/ods-ci/assets/88311595/108d516b-b542-4033-ab68-f04d0eb6a988)

Strip maybe can be avoided here, it was necessary in a previous version of the script.

In any case, without both it works, but output goes to new line when unnecessary